src/core/auth/ldap/ldap.go

```diff
@@ -117,24 +122,90 @@ func (l *Auth) attachLDAPGroup(ctx context.Context, ldapUsers []model.User, u *m
 		return
 	}
 	userGroups := make([]ugModel.UserGroup, 0)
+	if groupCfg.AttachParallel {
+		log.Debug("Attach LDAP group in parallel")
+		l.attachGroupParallel(ctx, ldapUsers, u)
```

Comment:
Seems Attach group sequentially is the "in parallel" case when worker number is 1, it would be better to converge these two logic branches?

Reply:
Per discussion Stone prefers keep the dup code in a separate branch before it's verified in customer's env. 
I'm resoving this comment.